cw-orch-daemon/src/sync/core.rs

```diff
@@ -202,3 +202,38 @@ impl TxHandler for Daemon {
         })
     }
 }
+
+impl WalletBalanceAssertion for Daemon {
```

Comment:
Imo this should be implemented for Wallet (i.e. the sender), not Daemon. 

`TxHandler` has an associated type `Sender`. I think that all sender-related logic should be encapsulated in traits that are a trait bound for this associated type. I.e. 

```rust
pub trait TxHandler: ChainState + Clone {

    type Sender: Clone + Bank ;
```

Then `Bank` could include a `.balance` and `.send_funds` method that's already in other places in this repo (imo send should be removed from the mock/daemon and only be available here) 

And then the assertion can be a supertrait of this `Bank` trait. I.e. 

```rust
trait BalanceAssertion: Bank {
// logic can be implemented here
}

// Blanket implementation for anything that implements Bank
impl <T: Bank> BalanceAssertion for <T> {
 // will be empty
}
```

Reply:
The question then becomes where we want to translate from the gas requirement to the actual fee requirement and in my opinion that should probably be done in the deploy trait because that's really where it's important. But that means that we need to somehow expose some endpoints on either the TxHandler trait that allows us to get the gas price for a specific environment. WDYT?